src/heretic/schemas.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+
+@dataclass
+class GenerationStep:
+    step_index: int
+    token_id: int
+    token: str
+    logprob: float | None
+    topk: List[Dict[str, float]] | None = None
+    entropy: float | None = None
+
+    # Optional internals
+    last_hidden_state: List[float] | None = None
+    residuals_per_layer: List[List[float]] | None = None
+    attention_summary: Dict[str, Any] | None = None
+
+
+@dataclass
+class GenerationTrace:
+    steps: List[GenerationStep]
+    finish_reason: str | None = None
+
+
+@dataclass
+class ContextMetadata:
+    # model and generation params
+    system_prompt: str | None = None
+    model_name: str | None = None
```

Comment:
These two aren't necessary. Plugins should have access to the `Settings`, which already contain that information and much more.